src/clients/checkCredit.js

```diff
@@ -0,0 +1,16 @@
+import Credit from "../models/credit.js";
+
+export default async (messageCost) => {
+
+    try {
+        const existingCredit = await Credit.findOne()
+
+        if ((existingCredit) && (existingCredit.amount - messageCost >= 0)) {
+            return true
+        } else {
+            return false
+        }
+    } catch (err) {
+        console.log("Error while checking balance", err);
+    }
+}
```

Comment:
You are naming this function as checkCredit when it cheks and updates the credit. It's even more confusing on other files when you call it 

If I read this code I might try to call this functions multiple times to check if I have credit, each time I would be losing money 😱